react/ProductContext.js

```diff
@@ -89,6 +89,7 @@ class ProductContext extends Component {
       params,
       params: { slug },
       client,
+      catalog : { refetch },
```

Comment:
I've tested this `refetch` directly and it didn't work. Did you manage to get it working?

Reply:
I tested it and works but we should to implement a loading mask when this method is executed in the ProductDetail component.

Reply:
The refetch method comes in the ProductQuery prop, i tested with the same flow in the SearchResult and it update the product information.

![image](https://user-images.githubusercontent.com/20094423/56920386-ace07b00-6a88-11e9-9d40-cc568a6fc927.png)

Reply:
If I'm not mistake, `productQuery` doesn't "exist", but `catalogQuery`. Did you verify if this `loading` comes from it (`catalogQuery`)?